### PR TITLE
Feature/#23 sharedresponseproperties

### PR DESCRIPTION
**Pull request by jeanmarc**
Implements feature 23 (adding "default" section to the stubconfig, to supply default delay (and some preparations to make httpStatusCode and response (body) optional in the future).
Includes some refactoring/code cleanup.

### CODE REVIEW

File: src/main/scala/castalia/actors/JsonEndpointActor.scala [hunk 1]
```diff
@@ -24,40 +22,30 @@ class JsonEndpointActor( myStubConfig: StubConfig) extends Actor with ActorLoggi
 
   override def receive: Receive = {
     case request: RequestMatch =>
-      log.debug("receive requestmatch")
```
Comment (DieBauer) [inline src/main/scala/castalia/actors/JsonEndpointActor.scala:30]: Why remove all these debug statements? when in production, log level should be >=Info.

File: src/main/scala/castalia/StubConfigParser.scala [hunk 2]
```diff
@@ -1,15 +1,36 @@
...
+import castalia.model.Model.{ResponseConfig, DefaultResponseConfig, StubConfig}
 
 
 object StubConfigParser {
 
   def parseStubConfig(jsonFile: String): StubConfig = {
-    JsonConverter.parseJson[StubConfig](jsonFile)
+    val initialConfig = JsonConverter.parseJson[StubConfig](jsonFile)
+    (initialConfig.default) match {
+      case (Some(default)) => StubConfig( initialConfig.endpoint, initialConfig.default, addDefaults(default, initialConfig.responses))
```
Comment (TimSoethout) [inline src/main/scala/castalia/StubConfigParser.scala:11]: something like `initialConfig.copy(defaults = addDefaults(default, initialConfig.responses))` should do the same

File: src/main/scala/castalia/StubConfigParser.scala [hunk 3]
```diff
@@ -1,15 +1,36 @@
...
   }
+
+  private def addDefaults( default: DefaultResponseConfig, responses: List[ResponseConfig]): List[ResponseConfig] =
+    (responses) match {
+      case (Nil) => Nil
+      case (first :: rest) => mix(default, first) :: addDefaults( default, rest)
+    }
+
+  private def mix(default: DefaultResponseConfig, response: ResponseConfig) = {
+    val delayOption = (default.delay, response.delay) match {
```
Comment (jeanmarc) [inline src/main/scala/castalia/StubConfigParser.scala:28]: rewrite to if response.delay.isPresent ... (or something)

File: src/main/scala/castalia/StubConfigParser.scala [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
+      case _ => initialConfig
+    }
+
   }
 
   def parseStubConfigs(jsonFiles: List[String]): List[StubConfig] = {
     jsonFiles.map(parseStubConfig(_))
   }
+
+  private def addDefaults( default: DefaultResponseConfig, responses: List[ResponseConfig]): List[ResponseConfig] =
```
Comment (TimSoethout) [inline src/main/scala/castalia/StubConfigParser.scala:21]: `reponses.map(config => mix(default, config))`

File: src/main/scala/castalia/StubConfigParser.scala [hunk 5]
```diff
@@ -1,15 +1,36 @@
...
+    (responses) match {
+      case (Nil) => Nil
+      case (first :: rest) => mix(default, first) :: addDefaults( default, rest)
+    }
+
+  private def mix(default: DefaultResponseConfig, response: ResponseConfig) = {
+    val delayOption = (default.delay, response.delay) match {
+      case (_, Some(delay)) => Some(delay)
+      case (default, _) => default
+    }
```
Comment (jeanmarc) [inline src/main/scala/castalia/StubConfigParser.scala:31]: bad naming of matched object

File: src/main/scala/castalia/StubConfigParser.scala (hunk 3 above)
Comment (TimSoethout) [inline src/main/scala/castalia/StubConfigParser.scala:28]: Something with `delayOption.orElse(...)`

**Pull request closed without merging**